src/sentry/issues/escalating_issues_alg.py

```diff
@@ -0,0 +1,113 @@
+    standard deviation over the average. This is to get an understanding of how high or low the variance
+    is relative to the data. The CV is then placed into an exponential equation that outputs
+    a multiplier inversely related to how high the cv is. The multiplier is bounded between 2 and 5. The
+    ceilings for the next week are all the same - which is the maximum number of events in an hour over the
+    past week multiplied by this multiplier. This calculation is to accound for bursty issues or those that
+    have a very high variance.
+    The final spike limit for each hour is set to the max of the bursty limit bound or the calculated limit.
+    :param data: Dict of Snuba query results - hourly data over past 7 days
+    :param start_time: datetime indicating the first hour to calc spike protection for
+    :return output: Dict containing a list of spike protection values
+    """
+
+    # output list of dictionaries
+    output: List[ForecastObject] = []
+
+    input_dates = [datetime.strptime(x, "%Y-%m-%dT%H:%M:%S%f%z") for x in data["intervals"]]
+    output_dates = [start_time + timedelta(days=x) for x in range(14)]
+
+    ts_data = data["data"]
+
+    # if data is empty return empty output
+    if len(ts_data) == 0 or len(input_dates) == 0:
+        return output
+
+    ts_max = max(ts_data)
+
+    # minimum ceiling for issues that don't have much data from the past week
+    floor_bound = 200
+
+    # if the max number of issues in an hour bucket is 30,
+    # a floor bound is used as the issue ceiling for the next week
+    if ts_max < 30:
+        for output_ts in output_dates:
+            output.append(
+                {"forecasted_date": output_ts.strftime("%Y-%m-%d"), "forecasted_value": floor_bound}
+            )
+        return output
+
+    # if we have less than a week's worth of data (new issue),
+    # set the threshold to 10x the max of the dataset to account for
+    # how the pattern of the issue will change over the first week
+    if len(ts_data) < 168:
+        for output_ts in output_dates:
+            output.append(
+                {"forecasted_date": output_ts.strftime("%Y-%m-%d"), "forecasted_value": ts_max * 10}
+            )
+        return output
+
+    # gather stats from the timeseries - average, standard dev
+    ts_avg = statistics.mean(ts_data)
+    ts_std_dev = statistics.stdev(ts_data)
+
+    # calculate cv to identify how high/low variance is
+    ts_cv = ts_std_dev / ts_avg
+
+    # multiplier determined by exponential equation - bounded between [2,5]
+    regression_multiplier = min(max(2, 5 * ((math.e) ** (-0.65 * ts_cv))), 5)
+
+    # first ceiling calculation
+    limit_v1 = ts_max * regression_multiplier
+
+    # This second multiplier corresponds to 5 standard deviations above the avg ts value
+    ts_multiplier = min(max((ts_avg + (5 * ts_std_dev)) / ts_avg, 5), 8)
```

Comment:
Is it possible `ts_avg` == 0?

Reply:
I don't think so. `ts_avg` is the mean of number of events a group has, which would not be 0. @npusarla could you please correct me if I'm misunderstanding?

Reply:
yeah `ts_avg` can't be 0 if it gets to this point in the algorithm